linkerd/identity/src/lib.rs

```diff
@@ -2,12 +2,49 @@
 #![forbid(unsafe_code)]
 
 mod credentials;
-mod local;
-mod name;
-
-pub use self::{
-    credentials::{Credentials, DerX509},
-    local::LocalId,
-    name::Name,
-};
-pub use linkerd_dns_name::InvalidName;
+
+pub use self::credentials::{Credentials, DerX509};
+
+/// An authenticated endpoint identity descriptor.
+///
+/// Practically speaking, this is a DNS-like identity string that matches an
+/// x509 DNS SAN. This will in the future be updated to support SPIFFE IDs as
+/// well.
+///
+/// This isn't restricted to TLS or x509 uses. An authenticated Id could be
+/// provided by, e.g., a JWT.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct Id(pub linkerd_dns_name::Name);
+
+// === impl Id ===
+
+impl std::str::FromStr for Id {
```

Comment:
it feels a bit sketchy to me that this newtype is intended to specifically represent an authenticated name, but can also freely be parsed from a random string anywhere? or am I missing something here?

Reply:
Ah I can clarify the comment. It's not intended to be proof of authenticated, but the kind of identity used for authentication